src/ddscxx/include/dds/core/policy/detail/TCorePolicyImpl.hpp

```diff
@@ -783,7 +783,8 @@ THistory<D> THistory<D>::KeepAll()
 template <typename D>
 THistory<D> THistory<D>::KeepLast(uint32_t depth)
 {
-    return THistory(dds::core::policy::HistoryKind::KEEP_LAST, depth);
+    return THistory(dds::core::policy::HistoryKind::KEEP_LAST,
+                    static_cast<int32_t>(depth));
```

Comment:
I casted this to `int32_t` here, since the depth is `int32_t` all the way till the C API, there may be a reason why it is this way!